packages/openapi3/test/xml-models.test.ts

```diff
@@ -0,0 +1,705 @@
+          },
+        },
+      },
+      required: ["id"],
+    });
+  });
+  it("provide the namespace and prefix using enum on model", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @nsDeclarations
+        enum Namespaces {
+          smp:"http://example.com/schema",
+        }
+
+        @ns(Namespaces.smp)
+        model Book {
+          id: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+        },
+      },
+      required: ["id"],
+      xml: {
+        namespace: "http://example.com/schema",
+        prefix: "smp",
+      },
+    });
+  });
+  it("provide the namespace and prefix using enum on model and properties", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+        @nsDeclarations
+        enum Namespaces {
+          smp:"http://example.com/schema",
+          ns2:"http://example.com/ns2",
+        }
+
+        @ns(Namespaces.smp)
+        model Book {          
+          id: string;
+          @ns(Namespaces.smp)
+          title: string;
+          @ns(Namespaces.ns2)
+          author: string;
+        };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        id: {
+          type: "string",
+        },
+        title: {
+          type: "string",
+          xml: {
+            namespace: "http://example.com/schema",
+            prefix: "smp",
+          },
+        },
+        author: {
+          type: "string",
+          xml: {
+            namespace: "http://example.com/ns2",
+            prefix: "ns2",
+          },
+        },
+      },
+      required: ["id", "title", "author"],
+      xml: {
+        namespace: "http://example.com/schema",
+        prefix: "smp",
+      },
+    });
+  });
+});
+describe("Array of primitive types", () => {
+  it("unwrapped tags array in the xmlBook model.", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+      @name("xmlBook")
+      model Book {
+        @unwrapped 
+        tags: string[];
+      };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        tags: {
+          type: "array",
+          items: {
+            type: "string",
+          },
+        },
+      },
+      required: ["tags"],
+      xml: {
+        name: "xmlBook",
+      },
+    });
+  });
+  it("wrapped tags array in the xmlBook model.", async () => {
+    const res = await oapiForModel(
+      "Book",
+      `
+      @name("xmlBook")
+      model Book {
+        @name("ItemsTags")
+        tags: string[];
+      };`
+    );
+
+    deepStrictEqual(res.schemas.Book, {
+      type: "object",
+      properties: {
+        tags: {
+          type: "array",
+          xml: {
+            name: "ItemsTags",
+            wrapped: true,
+          },
+          items: {
+            type: "string",
```

Comment:
the xml guide is actually not accurate here. it should have `xml.name: string`

```yaml
   Pet:
      type: "object"
      properties:
        tags:
          type: "array"
          xml:
            name: "ItemsTags"
            wrapped: true
          items:
            type: string
            xml:
              name: string
      xml:
        name: "XmlPet"
```

otherwise the node will not be serialized as `<string></string>` like the spec says